packages/flutter/test/material/app_test.dart

```diff
@@ -435,4 +435,96 @@ void main() {
+    // Mock the Window to explicitly report a light platformBrightness.
+    final TestWidgetsFlutterBinding binding = tester.binding;
+    binding.window.platformBrightnessTestValue = Brightness.light;
+
+    ThemeData appliedTheme;
+
+    await tester.pumpWidget(
+      MaterialApp(
+        theme: ThemeData(
+          brightness: Brightness.light
+        ),
+        darkTheme: ThemeData(
+          brightness: Brightness.dark,
+        ),
+        home: Builder(
+          builder: (BuildContext context) {
+            appliedTheme = Theme.of(context);
+            return const SizedBox();
+          },
+        ),
+      ),
+    );
+
+    expect(appliedTheme.brightness, Brightness.light);
+  });
+
+  testWidgets('MaterialApp uses dark theme when platformBrightness is dark', (WidgetTester tester) async {
+    // Mock the Window to explicitly report a dark platformBrightness.
+    final TestWidgetsFlutterBinding binding = tester.binding;
+    binding.window.platformBrightnessTestValue = Brightness.dark;
+
+    ThemeData appliedTheme;
+
+    await tester.pumpWidget(
+      MaterialApp(
+        theme: ThemeData(
+          brightness: Brightness.light
+        ),
+        darkTheme: ThemeData(
+          brightness: Brightness.dark,
+        ),
+        home: Builder(
+          builder: (BuildContext context) {
+            appliedTheme = Theme.of(context);
+            return const SizedBox();
+          },
+        ),
+      ),
+    );
+
+    expect(appliedTheme.brightness, Brightness.dark);
+  });
+
+  testWidgets('MaterialApp switches themes when the Window platformBrightness changes.', (WidgetTester tester) async {
+    // Mock the Window to explicitly report a light platformBrightness.
+    final TestWidgetsFlutterBinding binding = tester.binding;
+    binding.window.platformBrightnessTestValue = Brightness.light;
+
+    ThemeData themeBeforeBrightnessChange;
+    ThemeData themeAfterBrightnessChange;
+
+    await tester.pumpWidget(
+      MaterialApp(
+        theme: ThemeData(
+          brightness: Brightness.light
+        ),
+        darkTheme: ThemeData(
+          brightness: Brightness.dark,
+        ),
+        home: Builder(
+          builder: (BuildContext context) {
+            if (themeBeforeBrightnessChange == null) {
+              themeBeforeBrightnessChange = Theme.of(context);
+            } else {
+              themeAfterBrightnessChange = Theme.of(context);
+            }
+            return const SizedBox();
+          },
+        ),
+      ),
+    );
+
+    // Switch the platformBrightness from light to dark and pump the widget tree
+    // to process changes.
+    binding.window.platformBrightnessTestValue = Brightness.dark;
+    await tester.pumpAndSettle();
+
+    expect(themeBeforeBrightnessChange.brightness, Brightness.light);
+    expect(themeAfterBrightnessChange.brightness, Brightness.dark);
+  });
```

Comment:
Can you add more fallback behavior tests? e.g. light is used when window is dark and darkTheme is not supplied, light theme is not supplied, changing from a darkTheme to a light window still works because it fallbacks to ThemeData.fallback etc?

Reply:
Done